tests/rptest/tests/transactions_test.py

```diff
@@ -249,3 +252,70 @@ def change_static_member_test(self):
             assert kafka_error.code() == ck.cimpl.KafkaError.FENCED_INSTANCE_ID
 
         producer.abort_transaction()
+
+
+class UpgradeTransactionTest(RedpandaTest):
+    @cluster(num_nodes=2, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def check_parsing_test(self):
+        self.redpanda = RedpandaService(
+            self.test_context,
+            1,
+            extra_rp_conf={
+                "enable_idempotence": True,
+                "enable_transactions": True,
+                "transaction_coordinator_replication": 1,
+                "id_allocator_replication": 1,
+                "enable_leader_balancer": False,
+                "enable_auto_rebalance_on_node_add": False,
+            },
+            environment={"__REDPANDA_LOGICAL_VERSION": 5})
+
+        self.redpanda.start()
```

Comment:
Any reason to not use installer infra to upgrade from v22.2? Or if this is meant to be a general purpose upgrade test for transactions (i.e. not targeting a specific behavioral change), upgrade from `installer.highest_from_prior_feature_version(RedpandaInstaller.HEAD)`